src/mono/mono.proj

```diff
@@ -464,11 +464,21 @@
     <!-- Linux options -->
     <ItemGroup Condition="'$(TargetsLinux)' == true">
       <_MonoCFLAGS Include="-Wl,--build-id=sha1" />
+      <_MonoCFLAGS Include="-Wno-int-conversion" />
```

Comment:
same question as above, why does this need to be disabled?

Reply:
Building for `ppc64le` with clang-15 gives out `Wint-conversion` errors. I can reproduce the error if you require the exact error message.

Reply:
Error related to `Wint-conversion` is the following:
```
  [ 97%] Building C object mono/mini/CMakeFiles/monosgen-objects.dir/var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/artifacts/obj/_version.c.o
  /var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/mono/mono/mini/tramp-ppc.c:675:11: error: incompatible integer to pointer conversion initializing 'guint8 *' (aka 'unsigned char *') with an expression of type 'unsigned long' [-Wint-conversion]
                  guint8 *thunk = GET_MEMORY_SLOT_THUNK_ADDRESS((guint32*)(code - 32));
                          ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/mono/mono/mini/tramp-ppc.c:678:11: error: incompatible integer to pointer conversion initializing 'guint8 *' (aka 'unsigned char *') with an expression of type 'unsigned long' [-Wint-conversion]
                  guint8 *thunk = GET_MEMORY_SLOT_THUNK_ADDRESS((guint32*)(code - 4));
                          ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  2 errors generated.
```

Reply:
Related to this piece of code: https://github.com/dotnet/runtime/blob/42bebc934bc86d4b612190810cfbbcb0cb9f7d3d/src/mono/mono/mini/tramp-ppc.c#L673-L680 added by @akoeplinger

Reply:
The `GET_MEMORY_SLOT_THUNK_ADDRESS` macro returns a `guint64`, so I guess this simply needs a cast to `guint8 *`.  @alhad-deshpande can you look into this?

Reply:
Looks like this was fixed in https://github.com/dotnet/runtime/pull/78102